wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -99,6 +108,110 @@ def __init__(
         self._fail_count = 0
         self.pod_names = pod_names
         self.secret = secret
+        self._status_lock = Lock()
+        self._status = Status("running")
+        self._watch_job_thread = Thread(target=self._watch_job, daemon=True)
+        self._watch_pod_thread = Thread(
+            target=self._watch_pod, args=(pod_names[0],), daemon=True
+        )
+
+    def _set_status(self, status: Status) -> None:
+        """Set the run status."""
+        with self._status_lock:
+            self._status = status
+
+    def _watch_pod(self, pod_name: str) -> None:
+        """Handle event stream from rank 0 pod."""
+        w = watch.Watch()
+        try:
+            for event in w.stream(
+                self.core_api.list_namespaced_pod,
+                namespace=self.namespace,
+                field_selector=f"metadata.name={pod_name}",
+            ):
+                self._handle_pod_event(event)
```

Comment:
we should terminate the watch if the pod is a in terminal state to save some resources